consensus/system_contract/consensus.go

```diff
@@ -227,8 +227,31 @@ func (s *SystemContract) VerifyUncles(chain consensus.ChainReader, block *types.
 	return nil
 }
 
+// CalcBlocksPerSecond converts period to milliseconds and calculate blocks per second
+// For example: if Period = 250ms = 0.25s, then periodMs = 250
+func CalcBlocksPerSecond(periodMs uint64) uint64 {
+	blocksPerSecond := 1000 / periodMs // integer division, e.g. 1000/250 = 4
```

Comment:
`periodMs` can be `0`. It is for devnet mode that we don't use, but still safer to handle that case here.